gossipd/gossip.c

```diff
@@ -1011,6 +1011,32 @@ static struct io_plan *hand_back_peer(struct io_conn *conn,
 			  read_returning_gossipfd, rpeer);
 }
 
+static struct io_plan *disconnect_peer(struct io_conn *conn, struct daemon *daemon,
+				       const u8 *msg)
+{
+	struct pubkey id;
+ 	struct peer *peer;
+
+	if (!fromwire_gossipctl_peer_disconnect(msg, &id))
+		master_badmsg(WIRE_GOSSIPCTL_PEER_DISCONNECT, msg);
+
+	peer = find_peer(daemon, &id);
+	if (peer && peer->local) {
+		io_close(peer->local->conn);
+		msg = towire_gossipctl_peer_disconnect_reply(msg);
+		daemon_conn_send(&daemon->master, take(msg));
+	} else {
+		status_trace("release_peer: peer %s %s",
+			     type_to_string(trc, struct pubkey, &id),
+			     !peer ? "not found"
+			     : peer->local ? "already releasing"
```

Comment:
I think this particular case is impossible given the `if` condition?  If `peer` is non-NULL and `peer->local` is set then we would have gotten into the "then" branch of the `if` above.  Ignore this comment if this is for future-proofness though...

(A comment to the effect that "local peers are gossiping, non-local peers are owned by another daemon and are not gossiping" would be good too...)

Reply:
I will remove that additional redundant check